.github/workflows/python_compile_requirements.yml

```diff
@@ -32,3 +32,4 @@ jobs:
           title: Compile python requirements
           author: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
           assignees: fvankrieken, damonmcc, alexrichey, sf-dcp
+          branch: compile_python_reqs
```

Comment:
Ooh the other thing I might add - run nightly build on `push` to this branch. Though ideally we'd need to ensure that dev tags have been published before that happens. Need to think about it

Reply:
(think this is what you mentioned in standup today) happy to save this for a follow-up

Reply:
there's probably just a better way to organize these workflows. Something like
- `build` workflow call can take "all" as an input (currently in nightly build it's more duplicating logic in the build workflow)
- docker workflow publishing via workflow call ("test" workflow which builds images doesn't call the docker publish workflow)
- test workflow with just tests (and no docker image) via workflow call

Which then get called by
- nightly build (build all, run tests)
- pr tests (build docker images if needed, run tests)
- bump python packages (build docker images, build all products, run tests)